### PR TITLE
Delete old database from cache dir

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/android-media-library/src/main/java/com/adobe/marketing/mobile/media/MediaCollectionTracker.java [hunk 1]
```diff
@@ -1233,6 +1241,9 @@ class ErrorMessage {
             "Media tracker is not tracking the State passed into 'API:trackEvent(StateEnd)'.";
     static final String ErrTrackedStatesLimitReached =
             "Media tracker is already tracking maximum allowed states (10) per session.";
+    static final String ErrInvalidErrorId =
+            "ErrorId passed into 'API:trackError' is invalid. Please pass valid non-empty non-null"
+                    + " string for ErrorId.";
```
Comment (addb) [inline code/android-media-library/src/main/java/com/adobe/marketing/mobile/media/MediaCollectionTracker.java:1246]: Can we merge the error msg in one string?

File: code/android-media-library/src/androidTest/java/com/adobe/marketing/mobile/media/MediaDBServiceTests.java [hunk 2]
```diff
@@ -273,4 +275,26 @@ public void test_deleteAllHits_multipleSessionsPresent() {
...
+        oldDB.createNewFile();
+
+        File newDB = context.getDatabasePath(MEDIA_DATABASE);
+        newDB.delete();
+
+        // Before upgrade
+        assertTrue(oldDB.exists());
+        assertFalse(newDB.exists());
+
+        MediaDBService dbService = new MediaDBServiceImpl();
```
Comment (addb) [inline code/android-media-library/src/androidTest/java/com/adobe/marketing/mobile/media/MediaDBServiceTests.java:294]: Should we also cover the flow where we don't have existing oldDb. Make sure everything works fine.

File: code/android-media-library/src/main/java/com/adobe/marketing/mobile/media/MediaCollectionTracker.java (hunk 1 above)
Comment (praveek) [reply code/android-media-library/src/main/java/com/adobe/marketing/mobile/media/MediaCollectionTracker.java:1246]: The code formatter formats it this way and should be fine.

File: code/android-media-library/src/androidTest/java/com/adobe/marketing/mobile/media/MediaDBServiceTests.java (hunk 2 above)
Comment (praveek) [reply code/android-media-library/src/androidTest/java/com/adobe/marketing/mobile/media/MediaDBServiceTests.java:294]: The regular tests in the class already covers the scenario

**Review by addb — APPROVED** [review]

**Pull request merged**